src/main/java/org/spongepowered/api/service/context/ContextCalculator.java

```diff
@@ -22,36 +22,37 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.service.permission.context;
-
-import org.spongepowered.api.service.permission.Subject;
+package org.spongepowered.api.service.context;
 
 import java.util.Set;
 
 /**
- * Calculate the availability of contexts. These methods may be invoked
- * frequently, and therefore should be fast.
+ * Calculate the availability of {@link Context}s for a {@link Contextual}. 
+ * These methods may be invoked frequently, and therefore should be fast.
+ * 
  */
 public interface ContextCalculator {
```

Comment:
I'm not sure what you mean use generics? ContextsCalculator is already used in the Permission API nothing here is new it's just getting repurposed also for Economy as there's overlap in functionality that benefits both.